chatbox_tests/test_arsenic.py

```diff
@@ -1,59 +1,112 @@
+import asyncio
+import multiprocessing
 import os.path
+import time
 
 import arsenic
 import pytest
 import pytest_asyncio
+from arsenic import keys
 
 from chatbox_tests import mocked_database, arsenic_tests_helpers
-from src import helper_functions, chatbox_websocket_server
+from src import helper_functions, chatbox_websocket_server, enums
 
 
-class TestUser:
+class User:
     def __init__(self):
         self.chat_room = helper_functions.generate_random_string(10)
         self.user_name = helper_functions.generate_random_string(10)
         self.session: arsenic.Session = None
 
 
 @pytest_asyncio.fixture
-async def server():
+async def chatbox_server():
     config_relative_path = os.path.join('chatbox_tests', 'chatbox_tests_config.json')
     config_absolute_path = os.path.abspath(config_relative_path)
     data = helper_functions.read_config(config_absolute_path)
     address = data['address']['name']
     port = data['address']['port']
-    chatbox_server = chatbox_websocket_server.Server(data)
-    chatbox_server.chatbox_database = mocked_database.MockedDatabase()
-    await chatbox_server.start(address, port)
+    server = chatbox_websocket_server.Server(data)
+    server.chatbox_database = mocked_database.MockedDatabase()
+    await server.start(address, port)
 
     yield
-    chatbox_server.stop()
-    await chatbox_server.wait_stop()
+    server.stop()
+    await server.wait_stop()
+
+
+@pytest_asyncio.fixture
+async def http_server():
+    p = multiprocessing.Process(target=arsenic_tests_helpers.run_http_server)
+    p.start()
+
+    yield
+    p.terminate()
 
 
 @pytest_asyncio.fixture
 async def user():
-    client = TestUser()
-    client.session = await arsenic_tests_helpers.creaate_session()
-    await arsenic_tests_helpers.connect_user(client.user_name, client.chat_room, client.session)
+    client = User()
+    client.session = await arsenic_tests_helpers.creaate_session(client.chat_room, True)
 
     yield client
     await arsenic.stop_session(client.session)
 
 
 @pytest.mark.asyncio
-async def test_first_connection(server, user):
+async def test_open_chatbox_default_link(chatbox_server, http_server, user):
```

Comment:
It would be better to not use the "user" fixture which connects to a chat via chat name in the link in a test which checks if default links works.